runtime/jit_vm/ctsupport.cpp

```diff
@@ -720,4 +720,64 @@ jitGetDeclaringClassOfROMField(J9VMThread *vmStruct, J9Class *clazz, J9ROMFieldS
 	return currentClass;
 }
 
+void
+allMethodFromSignatureEndDo(J9MethodFromSignatureWalkState *state)
+{
+	J9JavaVM *vm = state->classLoaderWalkState.vm;
+	vm->internalVMFunctions->allClassLoadersEndDo(&state->classLoaderWalkState);
+}
+
+J9Method*
+allMethodFromSignatureNextDo(J9MethodFromSignatureWalkState *state)
+{
+	UDATA result = 0;
+
+	J9JavaVM *vm = state->classLoaderWalkState.vm;
+	J9ClassLoader *classLoader = vm->internalVMFunctions->allClassLoadersNextDo(&state->classLoaderWalkState);
+	while (NULL != classLoader) {
+		J9Class *clazz = vm->internalVMFunctions->internalFindClassUTF8(state->vmThread, reinterpret_cast<U_8 *>(state->className), state->classNameLength, classLoader, J9_FINDCLASS_FLAG_EXISTING_ONLY);
+		if (NULL != clazz) {
+			result = vm->internalVMFunctions->javaLookupMethod(state->vmThread, clazz, reinterpret_cast<J9ROMNameAndSignature *>(&state->nameAndSig), NULL, J9_LOOK_JNI | J9_LOOK_NO_JAVA);
+			if (0 != result) {
+				break;
+			}
+		}
+
+		classLoader = vm->internalVMFunctions->allClassLoadersNextDo(&state->classLoaderWalkState);
+	}
+
+	return reinterpret_cast<J9Method*>(result);
+}
+
+J9Method*
+allMethodFromSignatureStartDo(J9MethodFromSignatureWalkState *state, J9JavaVM *vm, UDATA flags, char *className, U_32 classNameLength, const char *methodName, U_32 methodNameLength, const char *methodSig, U_32 methodSigLength)
+{
+	state->className = className;
+	state->classNameLength = classNameLength;
+	state->nameAndSig.name = methodName;
+	state->nameAndSig.nameLength = methodNameLength;
+	state->nameAndSig.signature = methodSig;
+	state->nameAndSig.signatureLength = methodSigLength;
+
+	J9VMThread *currentThread = vm->internalVMFunctions->currentVMThread(vm);
+	state->vmThread = currentThread;
+
+	UDATA result = 0;
+
+	J9ClassLoader *classLoader = vm->internalVMFunctions->allClassLoadersStartDo(&state->classLoaderWalkState, vm, J9CLASSLOADERWALK_INCLUDE_DEAD);
+	while (NULL != classLoader) {
+		J9Class *clazz = vm->internalVMFunctions->internalFindClassUTF8(currentThread, reinterpret_cast<U_8 *>(className), classNameLength, classLoader, J9_FINDCLASS_FLAG_EXISTING_ONLY);
+		if (NULL != clazz) {
+			result = vm->internalVMFunctions->javaLookupMethod(currentThread, clazz, reinterpret_cast<J9ROMNameAndSignature *>(&state->nameAndSig), NULL, J9_LOOK_JNI | J9_LOOK_NO_JAVA);
```

Comment:
The field should be `J9ROMNameAndSignature nameAndSig;` so this `reinterpret_cast` is not necessary.

Reply:
I copied the implementation from here:
https://github.com/eclipse/openj9/blob/48b1439ec6b72c81881904767832ad8697d8cc40/runtime/jcl/common/java_lang_Class.cpp#L943-L967

There are more examples of us doing something like this. Looking at the definition of this struct:

https://github.com/eclipse/openj9/blob/48b1439ec6b72c81881904767832ad8697d8cc40/runtime/oti/j9nonbuilder.h#L638-L641

How does one initialize `J9ROMNameAndSignature`?

Reply:
Sorry, I mislead you: The field should be `J9NameAndSignature nameAndSig;` and the cast to `J9ROMNameAndSignature *` is appropriate if you include the flag `J9_LOOK_DIRECT_NAS`. See uses of `newInstanceImplNameAndSig` as defined [here](https://github.com/eclipse/openj9/blob/master/runtime/codert_vm/cache.c#L31).

Reply:
As discussed privately, `J9JNINameAndSignature` is an appropriate type for `nameAndSig` when `javaLookupMethod` is used with the `J9_LOOK_JNI` flag.